src/codegen/compile.ml

```diff
@@ -8171,12 +7941,18 @@ encountered during code generation, the other is determined
 dynamically by, e.g. message payload. The latter will vary with
 each payload to decode.
 
+The static type table and a type descriptor are stored in passive 
+data segments. Instead of absolute memory addresses, the static type 
+table in the data segment only contains relative offsets into type 
+descriptor. When loaded, these offsets are patched by static addresses 
+that point into the type descriptor.
+
 The known Motoko types are accumulated in a global list as required
```

Comment:
I think we actually need two global lists, one for Candid types and one for Persistenct types, since the intpretation is slightly differnt for each (one has dedicated blobs/tuples the other not).

Reply:
Thank you very much. Yes, there are two separate constructions.